### PR TITLE
[WIP] refactor: move arguments types to types.arguments 

### PR DESCRIPTION
**Pull request by scratchmex**
Trying to follow up this proposal: https://github.com/strawberry-graphql/strawberry/pull/489 at the point "Convert arguments to StrawberryArgument"

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tests/schema/test_mutation.py [hunk 1]
```diff
@@ -214,12 +213,10 @@ class Input:
     class Mutation:
         @strawberry.mutation
         def say(self, input: Input) -> str:
-            data = dataclasses.asdict(input)
-
-            if is_unset(data["name"]):
+            if is_unset(input.name):
```
Comment (BryceBeagle) [inline tests/schema/test_mutation.py:216]: Why this change?

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: tests/schema/test_mutation.py (hunk 1 above)
Comment (scratchmex) [reply tests/schema/test_mutation.py:216]: when you run `dataclass.asdict(input)`, for some strange reason (idk if this is intented) it initializes a new class for each type, that is, as `name` field is of the type `UNSET`, it initializes a new instance of the class `_Unset` and therefore the function `is_unset` is false. Basically, is the for the test case to pass as I did this change:
https://github.com/strawberry-graphql/strawberry/blob/429faeec5db2ce061fba208c150e36314c58294c/strawberry/types/arguments.py#L31-L33

and previously it was https://github.com/strawberry-graphql/strawberry/blob/ba6f79a6cb17efb4918551225bd6c39a53e117ef/strawberry/arguments.py#L29-L30

I did that change to avoid multiple instances of the constant `UNSET` which is actually an instance of the class `_Unset`

**Comment by patrick91** [comment]
same as #1237 I think we should revisit both PRs in future
